imageloading-core/src/main/java/dev/chrisbanes/accompanist/imageloading/AndroidDrawablePainter.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.chrisbanes.accompanist.imageloading
+
+import android.graphics.PorterDuff
+import android.graphics.drawable.Animatable
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.ColorDrawable
+import android.graphics.drawable.Drawable
+import android.os.Handler
+import android.os.Looper
+import android.view.View
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.BlendMode
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.asImageAsset
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.graphics.drawscope.drawCanvas
+import androidx.compose.ui.graphics.nativeCanvas
+import androidx.compose.ui.graphics.painter.ColorPainter
+import androidx.compose.ui.graphics.painter.ImagePainter
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.core.graphics.drawable.DrawableCompat
+import kotlin.math.roundToInt
+
+/**
+ * A [Painter] which draws an Android [Drawable]. Supports [Animatable] drawables.
+ */
+class AndroidDrawablePainter(
+    private val drawable: Drawable
+) : Painter() {
+    private val drawableSize = Size(
+        drawable.intrinsicWidth.toFloat(),
+        drawable.intrinsicWidth.toFloat()
+    )
+
+    private var invalidateTick by mutableStateOf(0)
+    private var startedAnimatable = drawable is Animatable && drawable.isRunning
+
+    private val handler by lazy { Handler(Looper.getMainLooper()) }
+
+    init {
+        drawable.callback = object : Drawable.Callback {
+            override fun invalidateDrawable(d: Drawable) {
+                // Update the tick so that we get re-drawn
+                invalidateTick++
+            }
+
+            override fun scheduleDrawable(d: Drawable, what: Runnable, time: Long) {
+                handler.postAtTime(what, time)
+            }
+
+            override fun unscheduleDrawable(d: Drawable, what: Runnable) {
+                handler.removeCallbacks(what)
+            }
+        }
+    }
+
+    override fun applyAlpha(alpha: Float): Boolean {
```

Comment:
@FloatRange?

Reply:
This is an override of a protected method. Not much point adding a `@FloatRange` here tbh